src/enclave/tls_session.h

```diff
@@ -417,7 +417,7 @@ namespace ccf
         else
         {
           LOG_TRACE_FMT(
-            "TLS {} on flush: {}", session_id, tls::error_string(r));
+            "TLS session {} error on flush: {}", session_id, -r);
```

Comment:
This return code is (a negation of) a return value from `SSL_get_error`, that's _not_ the same domain as `ERR_get_error`. We can pass the opaque numbers from the latter to `error_string` (a wrapper for `ERR_error_string_n`), but not from the former. Passing the former produces extremely unhelpful junk. There doesn't seem to be any helpful stringification for these errors, so its a case of finding the macro definitions in the OpenSSL source and working out what went wrong (or, better, calling `ERR_get_error` to flush the error stack and hope it shows something related, but that's a more awkward rewrite).